File: summer/WebContent/js/system/windows/controls/primitives/IHierarchicalVirtualizationAndScrollInfo.js
/**
 * IHierarchicalVirtualizationAndScrollInfo
 */

define(["dojo/_base/declare", "system/Type"], function(declare, Type){
	var IHierarchicalVirtualizationAndScrollInfo = declare("IHierarchicalVirtualizationAndScrollInfo", null,{
		constructor:function(){

		}
	});
	
	Object.defineProperties(IHierarchicalVirtualizationAndScrollInfo.prototype,{
		  
//		HierarchicalVirtualizationConstraints 
		Constraints:
		{
			get:function(){},
			set:function(value){}
		},
//		HierarchicalVirtualizationHeaderDesiredSizes 
		HeaderDesiredSizes:
		{
			get:function(){}
		},
//		HierarchicalVirtualizationItemDesiredSizes 
		ItemDesiredSizes:
		{
			get:function(){},
			set:function(value){}
		},
//		Panel 
		ItemsHost:
		{
			get:function(){}
		},
//		bool 
		MustDisableVirtualization:
		{
			get:function(){},
			set:function(value){}
		},
//		bool 
		InBackgroundLayout:
		{
			get:function(){},
			set:function(value){}
		}
	});
	
	IHierarchicalVirtualizationAndScrollInfo.Type = new Type("IHierarchicalVirtualizationAndScrollInfo", 
			IHierarchicalVirtualizationAndScrollInfo, [Object.Type], true);
	return IHierarchicalVirtualizationAndScrollInfo;
});

File: org.summer.view.widget/src/org/summer/view/widget/controls/IHierarchicalVirtualizationAndScrollInfo.java
package org.summer.view.widget.controls;


public interface IHierarchicalVirtualizationAndScrollInfo {

    HierarchicalVirtualizationConstraints Constraints 
    {
        get;
    } 

    HierarchicalVirtualizationHeaderDesiredSizes HeaderDesiredSizes
    { 
        get;
    }

    HierarchicalVirtualizationItemDesiredSizes ItemDesiredSizes
    {
        get;
    }

    Panel ItemsHost 
    {
        get;
    }

    boolean MustDisableVirtualization
    { 
        get;
    } 

    boolean InBackgroundLayout 
    {
        get ;
    } 
}
